clusterloader2/pkg/measurement/common/network-policy/network-policy-enforcement-latency.go

```diff
@@ -0,0 +1,567 @@
+			return err
+		}
+	}
+
+	return nps.createPermissionResources()
+}
+
+func (nps *networkPolicyEnforcementMeasurement) initializeMeasurement(config *measurement.Config) error {
+	if nps.framework != nil {
+		return fmt.Errorf("the %q is already started. Cannot start again", networkPolicyEnforcementName)
+	}
+
+	var err error
+	if nps.targetLabelKey, err = util.GetString(config.Params, "targetLabelKey"); err != nil {
+		return err
+	}
+
+	if nps.targetLabelValue, err = util.GetString(config.Params, "targetLabelValue"); err != nil {
+		return err
+	}
+
+	if nps.testClientNamespace, err = util.GetStringOrDefault(config.Params, "testClientNamespace", netPolicyTestNamespace); err != nil {
+		return err
+	}
+
+	if nps.baseline, err = util.GetBoolOrDefault(config.Params, "baseline", false); err != nil {
+		return err
+	}
+
+	if nps.testClientNodeSelectorKey, err = util.GetString(config.Params, "testClientNodeSelectorKey"); err != nil {
+		return err
+	}
+
+	if nps.testClientNodeSelectorValue, err = util.GetString(config.Params, "testClientNodeSelectorValue"); err != nil {
+		return err
+	}
+
+	nps.framework = config.ClusterFramework
+	nps.k8sClient = config.ClusterFramework.GetClientSets().GetClient()
+
+	namespaceList, err := nps.k8sClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Target namespaces are only those that have a predefined test prefix.
+	testNamespacePrefix := nps.framework.GetAutomanagedNamespacePrefix()
+	for _, ns := range namespaceList.Items {
+		if strings.HasPrefix(ns.GetName(), testNamespacePrefix) {
+			nps.targetNamespaces = append(nps.targetNamespaces, ns.GetName())
+		}
+	}
+
+	if len(nps.targetNamespaces) == 0 {
+		return fmt.Errorf("cannot initialize the %q, no namespaces with prefix %q exist", networkPolicyEnforcementName, testNamespacePrefix)
+	}
+
+	return nil
+}
+
+// createPermissionResources creates ServiceAccount, ClusterRole and
+// ClusterRoleBinding for the test client pods.
+func (nps *networkPolicyEnforcementMeasurement) createPermissionResources() error {
+	templateMap := map[string]interface{}{
+		"Name":      netPolicyTestClientName,
+		"Namespace": nps.testClientNamespace,
+	}
+
+	if err := nps.framework.ApplyTemplatedManifests(manifestsFS, serviceAccountFilePath, templateMap); err != nil {
+		return fmt.Errorf("error while creating serviceaccount: %v", err)
+	}
+
+	if err := nps.framework.ApplyTemplatedManifests(manifestsFS, clusterRoleFilePath, templateMap); err != nil {
+		return fmt.Errorf("error while creating clusterrole: %v", err)
+	}
+
+	if err := nps.framework.ApplyTemplatedManifests(manifestsFS, clusterRoleBindingFilePath, templateMap); err != nil {
+		return fmt.Errorf("error while creating clusterrolebinding: %v", err)
+	}
+
+	return nil
+}
+
+func (nps *networkPolicyEnforcementMeasurement) run(config *measurement.Config) error {
```

Comment:
Could we add some check to `run` that `setup` was run beforehand?

Reply:
Done. Used `nps.framework == nil` to check if setup was run before.